tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CaffeineCacheStats.java

```diff
@@ -47,24 +38,14 @@ private CaffeineCacheStats() {}
      * Callers should ensure that they have {@link Caffeine#recordStats() enabled stats recording}
      * {@code Caffeine.newBuilder().recordStats()} otherwise there are no cache metrics to register.
      *
-     * @deprecated use {@link #registerCache(TaggedMetricRegistry, Cache, String)}
-     *
      * @param registry metric registry
      * @param cache cache to instrument
      * @param name cache name
+     * @deprecated use {@link CaffeineCacheMetrics#registerCache(TaggedMetricRegistry, Cache, String)}
      */
     @Deprecated
     public static void registerCache(MetricRegistry registry, Cache<?, ?> cache, String name) {
```

Comment:
Should we remove support for non-tagged registries in 1.0?

Reply:
I'm not sure we're quite ready to completely remove the `MetricRegistry` usages quite yet. I'm also wondering which module & package should contain the `TaggedMetricRegistry` APIs.

The way I'm currently thinking is creating a 1.0 with Java 11 source/target compatibility and with all of these older APIs deprecated. Then cleanup the remaining consumers using deprecated types like all the original invocation event handlers and `MetricRegistry`, and then cutting 2.0 at that point.

Reply:
If we're renaming packages, should we use new maven coordinates as well to allow old and new to live side-by-side?

> Then cleanup the remaining consumers using deprecated types like all the original invocation event handlers and MetricRegistry, and then cutting 2.0 at that point.

Would we rename packages again from `v1` to `v2`?

We may want to avoid modifying `tritium-registry` given the breadth of dependencies, and the static state. Several consumers depend on data written into the SharedTaggedMetricRegistries singleton getting recorded by our server library.  I could see a case for extracting the metric registry API and implementation into a separate repository where we can retain java8 support longer.

Reply:
Right now this is mostly a side project, so wanted to throw this draft up for thoughts.

I'm using v1 as a placeholder right now just to get a sense of what the intended exported APIs should be. Definitely interested in thoughts on how best to approach the modules & packaging as I know there are pros/cons to whether to include a major version in fully qualified names. 

As you mentioned, the `TaggedMetricRegistry` & `MetricName` bits are the ones that will be most difficult to evolve and I'm open to pulling those out to a separate module/repo as needed.